teachingtoolshed/api/edstem.py

```diff
@@ -116,6 +162,112 @@ def get_lesson(self, lesson_id: int) -> Dict[str, Any]:
         lesson = self._ed_get_request(lesson_path)["lesson"]
         return lesson
 
+    def get_slide(self, slide_id: int) -> Dict[str, Any]:
+        """Gets metadata for a single slide. Endpoint: /lessons/slides/{slide_id}
+
+        Args:
+            slide_id: Identifier for slide
+
+        Returns:
+            A JSON object with the specified slide's metadata
+        """
+        slide_path = urljoin(EdStemAPI.API_URL, "lessons", "slides", slide_id)
+        slide = self._ed_get_request(slide_path)["slide"]
+        return slide        
+
+    def create_lesson(
+        self, 
+        title : str = None,
+        options : Dict[str, Any] = {}
+        ) -> Dict[str, Any]:
+        """Creates a new Ed lesson. Endpoint: /courses/{course_id}/lessons
+
+        Args:
+            title: Title for the new lesson
+            options: Dictionary of options to set on the lesson
+
+        Returns:
+            A JSON object with the new lesson's metadata
+        """
+        lessons_path = urljoin(EdStemAPI.API_URL, f"courses/{self._course_id}/lessons")
+        lesson_dict = {
+            "lesson": ({
+                "title" : title
+            } | options)
+        }
+        lesson = json.loads(self._ed_post_request(lessons_path, json=lesson_dict))["lesson"]
+        return lesson
+
+    def edit_lesson(
+        self, 
+        lesson_id : int,
+        options : Dict[str, Any] = {}
+        ) -> Dict[str, Any]:
+        """Modifies an existing Ed lesson. Endpoint: /lessons/{lesson_id}
+
+        Args:
+            lesson_id: Identifier for lesson
+            options: Dictionary of options to set on the lesson
+
+        Returns:
+            A JSON object with the updated lesson's metadata
+        """
+        lesson = self.get_lesson(lesson_id)
+        lesson_path = urljoin(EdStemAPI.API_URL, f"lessons/{lesson_id}")
+        lesson_dict = {
+            "lesson": lesson | options
+        }
+        lesson = json.loads(self._ed_put_request(lesson_path, json=lesson_dict))["lesson"]
+        return lesson
+
+    def clone_slide(
+        self,
+        slide_id: int,
+        lesson_id: int,
+        is_hidden: bool = False,
+        ) -> Dict[str, Any]:
+        """Clones an existing Ed slide into a new lesson. Endpoint: /lessons/slides/{slide_id}/clone
+
+        Args:
+            slide_id: Identifier for slide to clone
+            lesson_id: Identifier for lesson to clone into
+
+        Returns:
+            A JSON object with the cloned slide's metadata
+        """
+
+        clone_path = urljoin(EdStemAPI.API_URL, f"lessons/slides/{slide_id}/clone")
+        payload = {
+            "lesson_id": lesson_id,
+            "is_hidden": is_hidden
+        }
+        slide = json.loads(self._ed_post_request(clone_path, json=payload))["slide"]
+        return slide
+
+    # def edit_slide(
```

Comment:
What is the point of including this?

Reply:
I eventually want to be able to edit slides in the API (mostly to change titles), but it doesn't work yet. Fine to leave it out until I get it working.

Reply:
Let me know if you want me to change the PR. I think you can do it, though?

Reply:
Eh it's fine to leave it then.